File: 프로그래머스/기능개발/developefunction.cpp
#include <vector>
#include <iostream>
using namespace std;

vector<int> solution(vector<int> progresses, vector<int>speeds) {
	vector<int> answer;
	while (!progresses.empty()) {
		int t = (100 - progresses[0]) / speeds[0];
		int rest = (100 - progresses[0]) % speeds[0];
		if (rest != 0) {
			t += 1;
		}
		int curtotal=0;
		for (int i = 0; i < progresses.size();i++) {
			progresses[i] += speeds[i] * t;
		}
		int cur;
		for (cur = 0; cur < progresses.size(); cur++) {
			if (progresses[cur] < 100) {
				break;
			}
			curtotal++;
		}
		progresses.erase(progresses.begin(), progresses.begin() + cur);
		speeds.erase(speeds.begin(), speeds.begin() + cur);
		answer.push_back(curtotal);
	}
	return answer;
	/*
	while (!arry.empty()) {
		int curtotal =0;
		while (arry[0]<100) {
			for (int i = 0; i<arry.size();) {
				arry[i] += speeds[i];
				if (arry[i] >= 100) {
					if (i != 0) {
						arry.erase(arry.begin() + i);
						speeds.erase(speeds.begin() + i);
					}
					curtotal++;
				}
				else {
					i++;
				}
			}
		}
		answer.push_back(curtotal);
	}
	*/
}
int main() {
	solution({ 93,30,55 }, { 1,30,5 });
}

File: 프로그래머스/기능개발/developefunction.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        ArrayList<Integer> arr = new ArrayList<>();
        for(int i=0; i<progresses.length;){
            int tmpnum = 100-progresses[i];
            int value = tmpnum/speeds[i];
            int rest = tmpnum%speeds[i];
            if(rest ==0) value++;
            boolean check = true;
            int tempindex = i;
            int task =0;
            for(int j=i;j<progresses.length;j++){
                progresses[j] += speeds[j]*value;
                if(progresses[j]>=100&&check){
                    tempindex++;
                    task++;
                }
                else check=false;
            }
            arr.add(task);
            i = tempindex;
        }
        int[] answer = new int[arr.size()];
        for(int i=0; i<answer.length;i++)
            answer[i]= arr.get(i);
        return answer;
    }
}
public class test{
    public static void main(String[] args){
        Solution a = new Solution();
        int[] arr = {93, 30, 55};
        int[] arr2 = {1, 30, 5};
        int[] b = a.solution(arr,arr2);
    }
}
